File: 0007-10_001_prime/main.py
#!/usr/bin/env python

def is_prime(number: int) -> bool:
    if number < 2:
        return False

    for i in range(2, number - 1):
        if number % i == 0:
            return False

    return True


def main():
    LIMIT = 10_001
    current = 0
    number = 1
    prime = None

    while current < LIMIT:
        if is_prime(number):
            prime = number
            current += 1

        number += 1

    print(prime)


if __name__ == "__main__":
    main()

File: 0007-10_001_prime/main.rs
fn is_prime(number: i64) -> bool {
    if number < 2 {
        return false;
    }
    
    for i in 2..(number - 1) {
        if number % i == 0 {
            return false;
        }
    }
    
    return true;
}

fn main() {
    let limit: i64 = 10_001;
    let mut current: i64 = 0;
    let mut number: i64 = 1;
    let mut prime: i64 = 1;
    
    while current < limit {
        if is_prime(number) {
            prime = number;
            current += 1;
        }
        
        number += 1;
    }
    
    println!("{}", prime);
}
